### PR TITLE
Adds the basis for the protect command and nginx snippet

### PR DESCRIPTION
**Pull request by 3mkay**
Adds a command to enable/disable/reset basic auth on a project.

Assumptions:

1. The user is running the command on a Nixsal server. Otherwise it exits with a message
2. The username will always be the same as the logged-in user.
3. Password will be provided during set-up or reset.

### CODE REVIEW

**Comment by 3mkay** [comment]
Capturing this code here, so it doesn't get lost. Needs more testing before its ready to merge.

**Comment by bserem** [comment]
Some thoughts on this one:

1. I would prefer if the file wasn't there by default, but only if it is copied over with the `protect` command. So, have the `.conf` file under `scripts/nginx` and copy it. If it is in the nginx directory right away it's gonna be committed in the first place. If committed, then it will appear under changed files when edited, and folks will most probably commit it making a mess between nixsal and non-nixsal users.
2. `protect` sounds strange (to me?), `ddev htauth [enable|disable]` sounds better. But maybe `protect` makes more sense to people who never dealt with webserver configuration in the past.
3. This is important: we need to cover `apache` too. There are at least 2 projects running apache as their webserver, under DDEV.
4. Should allow user to setup a password, so that users use the same password in all their nixsal projects leading to less confusion?

**Comment by 3mkay** [comment]
> Some thoughts on this one:
> 
>     1. I would prefer if the file wasn't there by default, but only if it is copied over with the `protect` command. So, have the `.conf` file under `scripts/nginx` and copy it. If it is in the nginx directory right away it's gonna be committed in the first place. If committed, then it will appear under changed files when edited, and folks will most probably commit it making a mess between nixsal and non-nixsal users.

I guess we should probably exclude it using gitignore whether it's there by default or not. This is always going to be a local configuration option.

>     2. `protect` sounds strange (to me?), `ddev htauth [enable|disable]` sounds better. But maybe `protect` makes more sense to people who never dealt with webserver configuration in the past.

Naming things is hard. It's easier to remember 'protect' than 'htauth' or 'htpasswd' or whatever else.

>     3. This is important: we need to cover `apache` too. There are at least 2 projects running apache as their webserver, under DDEV.

It's important, but not urgent. If there are only a handful of sites from the 90+ then apache can come later, but definitely shouldn't be overlooked.

>     4. Should allow user to setup a password, so that users use the same password in all their nixsal projects leading to less confusion?

Ideally each nixsal box would have a global user:pass, but I've not dug into that yet. II feel it's outside of the remit for this command though.

**Comment by bserem** [comment]
I'm fully covered, apart from the excluding the file part. DDEV manages `.gitignore` files on it's own, and I'd like to keep it that way for now.

In any case, let's merge this and come back to it when we see it in action!

**Comment by 3mkay** [comment]
> I'm fully covered, apart from the excluding the file part. DDEV manages `.gitignore` files on it's own, and I'd like to keep it that way for now.

I was assuming we'd add it to the starterkit project `.gitignore` and each project as we come across it. Not sure whether that's what you mean or not?

> In any case, let's merge this and come back to it when we see it in action!

Indeed. I'll resolve the conflict and see what it looks like

**Comment by bserem** [comment]
> I was assuming we'd add it to the starterkit project `.gitignore` and each project as we come across it. Not sure whether that's what you mean or not?

DDEV creates `.gitignore` files in several directories when it starts.

May I ask that you add the command in the README file too? I can take care of this if you want

File: README.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ Highly opinionated set of configs and commands used by Annertech in our DDEV wor
 - - `cloudflare`: Shares the project with the outside world over a Cloudflare tunnel
 - - `devmode [on|off]`: Adds custom settings.local.php file and allows easy toggle between production and development mode
 - - `login`: Opens a browser and logs you in to Drupal (works on local environments only)
+- - `protect`: Enable or disable basic auth on a nixsal hosted dev project
```
Comment (bserem) [inline README.md:13]:
```suggestion
- - `protect [on|off|reset]`: Enable or disable basic auth on a nixsal hosted dev project
```

**Comment by bserem** [comment]
As far as I'm concerned you can merge this whenever you want, so that people test it on nixsal

**Pull request merged**